docs/querying/set-query-context.md

```diff
@@ -0,0 +1,310 @@
+
+To set query context via the HTTP API, You can start with your SQL query:
+
+   ```sql
+   SELECT COUNT(*) FROM data_source WHERE foo = 'bar' AND __time > TIMESTAMP '2000-01-01 00:00:00'
+   ```
+
+Then you need to create a JSON request with your query and context parameters. The following example sets the `sqlTimeZone` parameter:
+
+   ```json
+   {
+     "query" : "SELECT COUNT(*) FROM data_source WHERE foo = 'bar' AND __time > TIMESTAMP '2000-01-01 00:00:00'",
+     "context" : {
+       "sqlTimeZone" : "America/Los_Angeles"
+     }
+   }
+   ```
+
+You can then send the request to Druid's SQL API HTTP endpoint:
+
+   ```bash
+   curl -X POST 'http://localhost:8888/druid/v2/sql' \
+     -H 'Content-Type: application/json' \
+     -d '{
+       "query": "SELECT COUNT(*) FROM data_source WHERE foo = '\''bar'\'' AND __time > TIMESTAMP '\''2000-01-01 00:00:00'\''",
+       "context": {
+         "sqlTimeZone": "America/Los_Angeles"
+       }
+     }'
+   ```
+
+Druid will execute your query with the specified context parameters and return the results.
+
+You can set multiple context parameters in a single request:
+
+```json
+{
+  "query" : "SELECT COUNT(*) FROM data_source WHERE foo = 'bar'",
+  "context" : {
+    "timeout" : 30000,
+    "priority" : 100,
+    "useCache" : false,
+    "sqlTimeZone" : "America/Los_Angeles"
+  }
+}
+```
+For more information on how to format Druid SQL API requests and handle responses, see [Druid SQL API](../api-reference/sql-api.md).
+
+
+### JDBC driver API
+
+When connecting to Druid through JDBC, you set query context parameters as connection properties. This approach is useful when integrating Druid with BI tools or Java applications.
+
+Druid uses the Avatica JDBC driver (version 1.23.0 or later recommended). Note that Avatica does not support passing connection context parameters from the JDBC connection string—you must use a `Properties` object instead.
+
+**Basic example:**
+
+You can set query context parameters when creating your JDBC connection:
+
+```java
+// Connect to /druid/v2/sql/avatica/ on your Broker
+String url = "jdbc:avatica:remote:url=http://localhost:8888/druid/v2/sql/avatica/;transparent_reconnection=true";
+
+// Set any query context parameters you need here
+Properties connectionProperties = new Properties();
+connectionProperties.setProperty("sqlTimeZone", "America/Los_Angeles");
+connectionProperties.setProperty("useCache", "false");
+connectionProperties.setProperty("priority", "100");
+
+try (Connection connection = DriverManager.getConnection(url, connectionProperties)) {
+    try (
+        final Statement statement = connection.createStatement();
+        final ResultSet resultSet = statement.executeQuery("SELECT COUNT(*) FROM data_source WHERE foo = 'bar'")
+    ) {
+        while (resultSet.next()) {
+            // Process result set
+            System.out.println("Count: " + resultSet.getLong(1));
+        }
+    }
+}
+```
+
+**With authentication:**
```

Comment:
Suggest removing this if it's already covered; if you want, you can mention it in a line and link to the specific spot